storage/series.go

```diff
@@ -332,21 +333,27 @@ func (s *dbSeries) Bootstrap(rs m3db.DatabaseSeriesBlocks, cutover time.Time) er
 	// data accumulated during bootstrapping.
 	s.buffer.DrainAndReset(true)
 
+	// NB(xichen): if an error occurred during series bootstrap, we close
+	// the database series blocks and mark the series bootstrapped regardless
+	// in the hope that the other replicas will provide data for this series.
+	var err error
 	for i := range s.pendingBootstrap {
 		stream := s.pendingBootstrap[i].encoder.Stream()
-		err := s.drainStream(rs, stream, cutover)
+		err = s.drainStream(rs, stream, cutover)
 		stream.Close()
 		if err != nil {
-			s.Unlock()
-			return err
+			rs.Close()
+			rs = NewDatabaseSeriesBlocks(s.opts)
+			err = xerrors.NewRenamedError(err, fmt.Errorf("error occurred bootstrapping series %s: %v", s.seriesID, err))
+			break
```

Comment:
Was just worried the blocks that have been encoded might have been corrupted somehow due to the error, but I can close and recreate the block and continue instead of breaking out of the loop as suggested.